### PR TITLE
Rework flow monitoring topology to support parallelism

### LINKED ISSUE #3968: Flow SLA monitoring
Open-kilda should detect flows with violated latency SLA in real-time. 

To determine actual flow latency system should use Server42 latency measurement value if possible. In other case actual flow latency should be calculated as a sum of flow path links latencies with priority to round-trip ISL latency values over one-way ISL latency values.

### PR DESCRIPTION
**Pull request by rozdy**
Close #3968

### CODE REVIEW

**Review by dpoltavets — APPROVED** [review]

**Review by sergii-iakovenko — COMMENTED** [review]
Looks good. Minor comment left.

File: src-java/flowmonitoring-topology/flowmonitoring-storm-topology/src/main/java/org/openkilda/wfm/topology/flowmonitoring/bolt/FlowCacheBolt.java [hunk 1]
```diff
@@ -17,56 +17,72 @@
...
     private PersistenceManager persistenceManager;
     private Duration flowRttStatsExpirationTime;
+    private MetricFormatter metricFormatter;
 
     private transient FlowCacheService flowCacheService;
+    private transient CalculateFlowLatencyService calculateFlowLatencyService;
 
     public FlowCacheBolt(PersistenceManager persistenceManager, Duration flowRttStatsExpirationTime,
-                         String lifeCycleEventSourceComponent) {
+                         String prefix, String lifeCycleEventSourceComponent) {
```
[the remark this answers is not in the document]
Comment (rozdy) [reply src-java/flowmonitoring-topology/flowmonitoring-storm-topology/src/main/java/org/openkilda/wfm/topology/flowmonitoring/bolt/FlowCacheBolt.java]: Fixed.

File: src-java/flowmonitoring-topology/flowmonitoring-storm-topology/src/main/java/org/openkilda/wfm/topology/flowmonitoring/model/LinkState.java [hunk 2]
```diff
@@ -18,21 +18,38 @@
...
+        this.rttTimestamp = rttTimestamp;
+        this.oneWayLatency = Duration.ofNanos(oneWayLatency);
+    }
 
     /**
      * Get most recent latency for the link.
      */
-    public long getLatency(long currentTimestamp, long rttLatencyExpiration) {
-        if (currentTimestamp > rttTimestamp + rttLatencyExpiration) {
+    public Duration getLatency(Instant instant, Duration rttLatencyExpiration) {
```
Comment (sergii-iakovenko) [inline src-java/flowmonitoring-topology/flowmonitoring-storm-topology/src/main/java/org/openkilda/wfm/topology/flowmonitoring/model/LinkState.java:41]: Will it make sense to rename "instant" to "now"?

**Review by sergii-iakovenko — APPROVED** [review]
Looks good. Minor comment left.

File: src-java/flowmonitoring-topology/flowmonitoring-storm-topology/src/main/java/org/openkilda/wfm/topology/flowmonitoring/model/LinkState.java (hunk 2 above)
Comment (rozdy) [reply src-java/flowmonitoring-topology/flowmonitoring-storm-topology/src/main/java/org/openkilda/wfm/topology/flowmonitoring/model/LinkState.java]: Renamed.

**Pull request merged**